pkg/prom/query.go

```diff
@@ -171,6 +176,16 @@ func (ctx *Context) RawQuery(query string) ([]byte, error) {
 	u := ctx.Client.URL(epQuery, nil)
 	q := u.Query()
 	q.Set("query", query)
+
+	// for non-range queries, we set the timestamp for the query to time-offset
+	// this is a special use case that's typically only used when our primary
+	// prom db has delayed insertion (thanos, cortex, etc...)
+	if promQueryOffset != 0 && ctx.name != AllocationContextName {
+		q.Set("time", time.Now().Add(-promQueryOffset).UTC().Format(time.RFC3339))
+	} else {
+		q.Set("time", time.Now().UTC().Format(time.RFC3339))
+	}
```

Comment:
Right now, this is a complete hack, but since it's isolated (this config isn't exposed on the cost-model, just the agent in the helm chart), I'm punting because I eventually want to use a remote write, which will have current time metric entry. For whatever reason, whenever we add the 3hour query offset for allocation queries, ETL fails to build. No rhyme or reason. Going to have to give it another look later.

Reply:
@mbolt35 so Asset and Allocation are out-of-sync by 3h for hosted?